redwood-treehouse-guest/src/commonMain/kotlin/app/cash/redwood/treehouse/treehouseCompose.kt

```diff
@@ -63,7 +67,21 @@ private class RedwoodZiplineTreehouseUi(
     )
     this.composition = composition
 
-    composition.bind(treehouseUi, uiConfigurations)
+    this.saveableStateRegistry = SaveableStateRegistry(
+      restoredValues = stateSnapshot?.toValuesMap(),
+      canBeSaved = { true },
```

Comment:
TODO: limit this to the stuff we can save

Reply:
probably add a `stateSnapshot.canBeSaved` func